### PR TITLE
feat: add PRINTR bonding curve liquidity source

### PR DESCRIPTION
**Pull request by 0xflippr**
## Summary

- Add `pkg/liquidity-source/printr/` — a new liquidity source for PRINTR's pre-graduation bonding curve tokens
- Replicate Solidity constant-product bonding curve math (`CalcBuyCost`, `CalcBuyTokenAmount`, `CalcSellRefund`) in Go using `holiman/uint256`
- Implement `IPoolSimulator` (CalcAmountOut, CalcAmountIn, UpdateBalance, CloneState), `IPoolsListUpdater` (HTTP tokenlist API), and `IPoolTracker` (RPC multicall)

## Details

PRINTR tokens trade on a constant-product bonding curve before graduating to DEX liquidity pools. This integration enables KyberSwap to route through these pre-graduation curves.

**Architecture:**
- Pool discovery via PRINTR's tokenlist API at `https://api-preview.printr.money/chains/{chainId}/tokenlist.json` (Uniswap tokenlist standard)
- State tracking via batched RPC: `getCurve(token)`, `tradingFee()`, `paused()`
- Graduated tokens (`completionThreshold == 0`) are excluded from routing
- Same proxy address on all EVM chains: `0xb77726291b125515d0a7affeea2b04f2ff243172`

**Math fidelity:**
- Integer division rounding matches Solidity exactly (round-up on buy cost, round-down on sell refund)
- Min fee = 1 wei when `tradingFee > 0` and amount > 0
- Completion threshold caps buy amounts
- Division-by-zero guards for edge cases

**Files:** 12 files, ~2k lines, 42 unit tests

## Test plan

- [x] `go build ./pkg/liquidity-source/printr/...`
- [x] `go vet ./pkg/liquidity-source/printr/...`
- [x] `go test -v -count=1 ./pkg/liquidity-source/printr/...` — 42/42 pass
- [ ] Cross-verify math outputs against Solidity `_estimateTokenCost` / `_estimateTokenRefund` with matching inputs

### CODE REVIEW

File: pkg/liquidity-source/printr/pool_list_updater.go [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+		if token.ChainId != int(u.config.ChainId) {
+			continue
+		}
+
+		isGraduated, _ := token.Extensions["isGraduated"].(bool)
+		if isGraduated {
+			continue
+		}
+
+		basePair, _ := token.Extensions["basePair"].(string)
```
Comment (NgoKimPhu) [inline pkg/liquidity-source/printr/pool_list_updater.go]: the api doesn't seem to have this?
